substrate/utils/frame/benchmarking-cli/src/storage/read.rs

```diff
@@ -60,13 +79,28 @@ impl StorageCmd {
 				_ => {
 					// regular key
 					let start = Instant::now();
-					let v = client
-						.storage(best_hash, &key)
+
+					let v = backend
+						.storage(key.0.as_ref())
 						.expect("Checked above to exist")
 						.ok_or("Value unexpectedly empty")?;
-					record.append(v.0.len(), start.elapsed())?;
+					record.append(v.len(), start.elapsed())?;
 				},
 			}
+			read_in_batch += 1;
+			if read_in_batch >= self.params.batch_size {
```

Comment:
I don't get the purpose of that? What should that achieve here?

Reply:
Using a new recorder for every read vs using the same for the entire batch produced significant different results, can be seen in the results I posed for `With TrieCache, With PoV Recorder` for different batch sizes.

 Since we use a single recorder per block not for every key, this tries to realistically compute the amortised cost of read, by creating new objects only every read.

Reply:
Ahh yeah makes sense! A small comment would be nice :)